implementations/rust/ockam/ockam_vault/src/lib.rs

```diff
@@ -46,8 +45,43 @@ mod software;
 /// Main vault types: PublicKey, Secret, SecretAttributes etc...
 mod types;
 
-pub use constants;
 pub use error::*;
 pub use software::*;
 pub use traits::*;
 pub use types::*;
+
+/// Feature set compatibility checks
+
+#[cfg(all(
+    feature = "disable_default_noise_protocol",
+    not(feature = "OCKAM_XX_25519_AES256_GCM_SHA256"),
+    not(feature = "OCKAM_XX_25519_ChaChaPolyBLAKE2s"),
+    not(feature = "OCKAM_XX_25519_AES128_GCM_SHA256")
+))]
+compile_error! {"NOISE protocol name not selected, please enable on of the following features: \"OCKAM_XX_25519_ChaChaPolyBLAKE2s\", \"OCKAM_XX_25519_AES128_GCM_SHA256\", \"OCKAM_XX_25519_AES256_GCM_SHA256\""}
```

Comment:
I think it would be more explicit/direct to have 3 exclusive features and declare `OCKAM_XX_25519_AES256_GCM_SHA256` as the default one). Otherwise we have to think in terms of double negation later in the code: 
```
if #[cfg(any(not(feature = "disable_default_noise_protocol")
```

But maybe I missed something that makes this approach problematic?

One thing I find unfortunate is that there's no `oneof` exclusive choice so we still have to write
```
#[cfg(not(any(
    feature = "OCKAM_XX_25519_AES256_GCM_SHA256",
    feature = "OCKAM_XX_25519_AES128_GCM_SHA256"
    feature = "OCKAM_XX_25519_ChaChaPolyBLAKE2s",
    )))]
compile_error! {"please select exactly one noise protocol"}

#[cfg(all(
    feature = "OCKAM_XX_25519_AES256_GCM_SHA256",
    any(
        feature = "OCKAM_XX_25519_ChaChaPolyBLAKE2s",
        feature = "OCKAM_XX_25519_AES128_GCM_SHA256"
    ))]
compile_error! {"only one protocol can be selected"}

#[cfg(all(
    feature = "OCKAM_XX_25519_AES128_GCM_SHA256"
    any(
        feature = "OCKAM_XX_25519_ChaChaPolyBLAKE2s",
        feature = "OCKAM_XX_25519_AES256_GCM_SHA256",
    ))]
compile_error! {"only one protocol can be selected"}

#[cfg(all(
    feature = "OCKAM_XX_25519_ChaChaPolyBLAKE2s"
    any(
        feature = "OCKAM_XX_25519_AES128_GCM_SHA256",
        feature = "OCKAM_XX_25519_AES256_GCM_SHA256",
    ))]
compile_error! {"only one protocol can be selected"}
```